File: Database Management Systems/Generic Database Binding/GenericDatabaseBinding/Observable.cs
using System.Collections.Generic;


namespace WindowsFormsApplication2 {

    public class Observable
	{
		private readonly List<Observer> _observers;

		public Observable()
		{
            _observers = new List<Observer>();
		}

		public void AddObserver(Observer observer)
		{
			_observers.Add(observer);
		}

		public void NotifyObservers(NotificationType type)
		{
			foreach (var observer in _observers)
			{
				observer.notifyMe(type);
			}
		}
	}
}

File: Systems for Design and Implementation/event.manager.java/src/main/java/network/Observable.java
package network;

import service.Notification;

import java.util.ArrayList;
import java.util.List;

/**
 * @author Alexandru Stoica
 * @version 1.0
 */

public class Observable {

    private List<Observer> observers;

    protected Observable(){
        observers = new ArrayList<>();
    }

    public void addObserver(Observer observer) {
        observers.add(observer);
    }

    public void removeObserver(Observer observer) {
        observers.remove(observer);
    }

    public void notifyObservers(Notification notification) {
        observers.forEach(observer -> observer.update(notification));
    }
}
